doctr/models/preprocessor/tensorflow.py

```diff
@@ -113,13 +114,13 @@ def __call__(self, x: Union[tf.Tensor, np.ndarray, List[Union[tf.Tensor, np.ndar
 
         elif isinstance(x, list) and all(isinstance(sample, (np.ndarray, tf.Tensor)) for sample in x):
             # Sample transform (to tensor, resize)
-            samples = list(multithread_exec(self.sample_transforms, x))
+            samples = list(multithread_exec(self.sample_transforms, x, threads=1 if self._runs_on_cuda else None))
```

Comment:
For the inference pipeline we can use threading because on CPU it works without issues :) 
So only disable if it runs on GPU